pkg/api/utils.go

```diff
@@ -63,3 +64,9 @@ func ValidateAndNormalizeEmail(email string) (string, error) {
 
 	return e.Address, nil
 }
+
+func (hs *HTTPServer) injectSpan(c *contextmodel.ReqContext, name string) (*contextmodel.ReqContext, trace.Span) {
+	ctx, span := hs.tracer.Start(c.Req.Context(), name)
+	c.Req = c.Req.WithContext(ctx)
+	return c, span
+}
```

Comment:
Would it make more sense to add a middleware instead or do we want to keep it only for specific endpoints? Do we also want to add other information we can gather to the span as well?

Reply:
Right now we want to keep it limited to specific endpoints and figure out our usecase. We can bolt onto this as we go and think about what a more generic middleware might look like. Having a span on api failures would be nice.